### PR TITLE
COM-12461: Create an AAC test suite generator

### PR DESCRIPTION
**Pull request by mdimopoulos**
- created new AAC test suite generator gen_aac.py
- tested to work with ISO_IEC_13818-4_2004 ADTS suite

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -43,7 +43,7 @@ packages = ["fluster", "fluster.decoders"]
 
 [tool.setuptools.data-files]
 "share/fluster/test_suites/aac" = [
-    "test_suites/aac/ISO_IEC_13818-4_2004.json",
```
Comment (mdimopoulos) [inline pyproject.toml:46]: This change here will break BC, but it's necessary to differentiate against suites that we are going to add soon. I think that we can do a new fluster release before the end of the year where we can include all BC breaking code and inform the users in a CHANGELOG about it.
As discussed the other day.
@rsanchez87 @rgonzalezfluendo thoughts?

Comment (rgonzalezfluendo) [reply pyproject.toml:46]: 100% agree

Comment (rsanchez87) [reply pyproject.toml:46]: Totally agree :+1:

**Review by rsanchez87 — CHANGES_REQUESTED** [review]
Here there is an exception when executing (**python3 scripts/gen_aac.py**):

```
Traceback (most recent call last):
  File "/home/rsanchez/SOFTWARE/fluster/scripts/gen_aac.py", line 330, in <module>
    generator.generate(not args.skip_download, args.jobs)
  File "/home/rsanchez/SOFTWARE/fluster/scripts/gen_aac.py", line 241, in generate
    raise Exception(f"Bitstream file {test_vector.input_file} not found in {dest_dir}")
Exception: Bitstream file al12_48.adts not found in resources/MPEG2_AAC-ADTS/al12_48
```

File: scripts/gen_aac.py [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+
+        for test_vector in test_suite.test_vectors.values():
+            dest_dir = os.path.join(test_suite.resources_dir, test_suite.name, test_vector.name)
+            dest_path = os.path.join(dest_dir, os.path.basename(test_vector.source))
+            absolute_input_path = os.path.join(pathlib.Path(__file__).parent.resolve(), dest_dir,
+                                               test_vector.input_file)
+
+            # Check that bitstream file is located inside the corresponding test vector folder
+            if not os.path.isfile(absolute_input_path):
+                raise Exception(f"Bitstream file {test_vector.input_file} not found in {dest_dir}")
```
Comment (rsanchez87) [inline scripts/gen_aac.py:241]: Here, exception

**Comment by mdimopoulos** [comment]
> There is an exception when executing (**python3 scripts/gen_aac.py**):
> 
> ```
> Traceback (most recent call last):
>   File "/home/rsanchez/SOFTWARE/fluster/scripts/gen_aac.py", line 330, in <module>
>     generator.generate(not args.skip_download, args.jobs)
>   File "/home/rsanchez/SOFTWARE/fluster/scripts/gen_aac.py", line 241, in generate
>     raise Exception(f"Bitstream file {test_vector.input_file} not found in {dest_dir}")
> Exception: Bitstream file al12_48.adts not found in resources/MPEG2_AAC-ADTS/al12_48
> ```
> 
> The reason why I'm launching from:
> 
> `user: fluster$ python3 scripts/gen_aac.py`
> 
> @mdimopoulos What do you think about not including the absolute path? (In the future we may have a CI and these commands will be executed from the root path of **fluster**...)
> 
> _In the rest of ts, this is the way it is._ In this branch from this one, it's working (**COM-12460**)

I have fixed this in the latest fixup commit.
In other test suites `absolute_input_path` is an absolute path. So for now i will leave it as is here too. In the future we can think about a refactoring of this logic in all generator scripts. It would be nice that all test suites are downloaded within a specific relative folder path always, independent of the current working directory. And we can also change absolute to relative paths by using the script's path at runtime (os.path.XYZ(__file__))

**Review by rsanchez87 — APPROVED** [review]
LGTM!

**Pull request merged**